### PR TITLE
restore: add a new interface for SplitClient to return the original region in BatchSplitRegions

### PR DESCRIPTION
**Pull request by glorv**
Signed-off-by: glorv <glorvs@163.com>

### What problem does this PR solve? 
- Add a new interface `SplitClient.BatchSplitRegionsRaw` to return the split region after batch split.
- Add a warn log if scatters region failed, so we can determine that region distributes unevenly is due to scattering fails.

### What is changed and how it works?
Return the split region in `BatchSplitRegion` result.

Because this change is incompatible, and directly change the original interface will cause build failure, so add a new method instead to be backward compatible.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test
 - No code

Code changes
 - Has interface methods change

Side effects

Related changes

### Release Note

 - No release note

### CODE REVIEW

File: pkg/restore/split_client.go [hunk 1]
```diff
@@ -51,6 +51,8 @@ type SplitClient interface {
 	// BatchSplitRegions splits a region from a batch of keys.
 	// note: the keys should not be encoded
 	BatchSplitRegions(ctx context.Context, regionInfo *RegionInfo, keys [][]byte) ([]*RegionInfo, error)
+	// BatchSplitRegionsRaw splits a region from a batch of keys and return the original region and split new regions
+	BatchSplitRegionsRaw(ctx context.Context, regionInfo *RegionInfo, keys [][]byte) (*RegionInfo, []*RegionInfo, error)
```
Comment (kennytm) [inline pkg/restore/split_client.go:55]: maybe better call it "BatchSplitRegionsWithOriginal" than calling it "raw".

**Review by kennytm — COMMENTED** [review]
rest LGTM

**Review by 3pointer — APPROVED** [review]
LGTM

**Pull request merged**